src/main/java/edu/cornell/kfs/module/ar/service/impl/CuContractsGrantsInvoiceCreateDocumentServiceImpl.java

```diff
@@ -179,23 +189,253 @@ protected boolean isExpenditureSubFund(String subFundGroupCode) {
         }
         return false;
     }
+
+    private Date determineLastBilledDateByInvoicingOption(List<ContractsAndGrantsBillingAwardAccount> awardAccounts, String invoicingOptionCode, Date awardLastBilledDate) {
+        Date computedLastBilledDate = null;
+        if (StringUtils.equalsIgnoreCase(CuArConstants.AwardInvoicingOptionCodeToName.INV_ACCOUNT.getCode(), invoicingOptionCode)
+                || StringUtils.equalsIgnoreCase(CuArConstants.AwardInvoicingOptionCodeToName.INV_CONTRACT_CONTROL_ACCOUNT.getCode(), invoicingOptionCode)) {
+            ContractsAndGrantsBillingAwardAccount accountToUse = null;
+            for (ContractsAndGrantsBillingAwardAccount awardAccount : awardAccounts) {
+                if (ObjectUtils.isNotNull(accountToUse)) {
+                    if (isNotExpenditureAccount(awardAccount)
+                            && ObjectUtils.isNotNull(awardAccount.getCurrentLastBilledDate())
+                            && ObjectUtils.isNotNull(accountToUse.getCurrentLastBilledDate())
+                            && awardAccount.getCurrentLastBilledDate().after(accountToUse.getCurrentLastBilledDate())) {
+                        accountToUse = awardAccount;
+                    }
+                } else if (isNotExpenditureAccount(awardAccount)) {
+                    accountToUse = awardAccount;
+                }
+            }
+            if (ObjectUtils.isNotNull(accountToUse)) {
+                computedLastBilledDate = accountToUse.getCurrentLastBilledDate();
+            } else {
+                LOG.error("determineLastBilledDateByInvoicingOption: NO award accounts passed to method when award had invoice option of Account or Contract Control Account. lastBilledDate being returned as computedLastBilledDate =" + computedLastBilledDate);
```

Comment:
I am not sure if it matters, but there are two conditions here, either no award accounts were passed in, or non the award accounts met the conditional logic